packages/react-devtools-shared/src/backend/views/Highlighter/index.js

```diff
@@ -151,14 +151,23 @@ export default function setupHighlighter(
     event.preventDefault();
     event.stopPropagation();
 
-    selectFiberForNode(((event.target: any): HTMLElement));
+    selectFiberForNode(
+      (((event.composed
+        ? event.composedPath()[0]
+        : event.target): any): HTMLElement),
+    );
   }
 
-  function onPointerOver(event: MouseEvent) {
+  let lastHoveredNode: HTMLElement | null = null;
+  function onPointerMove(event: MouseEvent) {
     event.preventDefault();
     event.stopPropagation();
 
-    const target = ((event.target: any): HTMLElement);
+    const target: HTMLElement = ((event.composed
+      ? event.composedPath()[0]
+      : event.target): any);
```

Comment:
Should we add some check that `target` != null?

Reply:
the null check wont be necessary in my opinion as the above code expression
```
(event.composed ? event.composedPath()[0] : event.target
```
is already responsible for checking the occurrence  of null or undefined value